--- origsrc/gnome-desktop-sharp-2.26.0/configure.in	2009-03-02 09:22:58.000000000 -0600
+++ src/gnome-desktop-sharp-2.26.0/configure.in	2009-12-09 02:51:21.302667000 -0600
@@ -5,7 +5,7 @@ AM_INIT_AUTOMAKE(gnome-desktop-sharp, 2.
 AM_MAINTAINER_MODE
 
 case "$host" in
-	*-*-mingw*|*-*-cygwin*)
+	*-*-mingw*)
 		platform_win32=yes
 		AC_DEFINE(PLATFORM_WIN32,1,[Platform is Win32])
 		CC="gcc -mno-cygwin -g"
@@ -99,13 +99,23 @@ if test "x$RUNTIME" != "no" ; then
 fi
 
 AC_PATH_PROG(CSC, mcs, no)
-if test `uname -s` = "Darwin"; then
-	LIB_PREFIX=
+case `uname -s` in
+  Darwin)
+	LIB_PREFIX=lib
+	LIB_MIDDLE=.
 	LIB_SUFFIX=.dylib
-else
-	LIB_PREFIX=.so
+	;;
+  CYGWIN*)
+	LIB_PREFIX=cyg
+	LIB_MIDDLE=-
+	LIB_SUFFIX=.dll
+	;;
+  *)
+	LIB_PREFIX=lib
+	LIB_MIDDLE=.so.
 	LIB_SUFFIX=
-fi
+	;;
+esac
 else
 AC_PATH_PROG(CSC, csc.exe, no)
 GACUTIL_FLAGS=
@@ -131,6 +141,7 @@ AC_SUBST(CSC)
 AC_SUBST(GACUTIL)
 AC_SUBST(GACUTIL_FLAGS)
 AC_SUBST(LIB_PREFIX)
+AC_SUBST(LIB_MIDDLE)
 AC_SUBST(LIB_SUFFIX)
 AC_SUBST(GENERATED_SOURCES)
 
--- origsrc/gnome-desktop-sharp-2.26.0/gnomedesktop/gnomedesktop-sharp.dll.config.in	2009-02-23 12:01:27.000000000 -0600
+++ src/gnome-desktop-sharp-2.26.0/gnomedesktop/gnomedesktop-sharp.dll.config.in	2009-12-09 02:51:21.302667000 -0600
@@ -1,7 +1,7 @@
 <configuration>
-  <dllmap dll="libglib-2.0-0.dll" target="libglib-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
-  <dllmap dll="libgobject-2.0-0.dll" target="libgobject-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
-  <dllmap dll="libgthread-2.0-0.dll" target="libgthread-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
-  <dllmap dll="libgtk-win32-2.0-0.dll" target="libgtk-x11-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
-  <dllmap dll="gnome-desktop-2" target="libgnome-desktop-2@LIB_PREFIX@.11@LIB_SUFFIX@"/>
+  <dllmap dll="libglib-2.0-0.dll" target="@LIB_PREFIX@glib-2.0@LIB_MIDDLE@0@LIB_SUFFIX@"/>
+  <dllmap dll="libgobject-2.0-0.dll" target="@LIB_PREFIX@gobject-2.0@LIB_MIDDLE@0@LIB_SUFFIX@"/>
+  <dllmap dll="libgthread-2.0-0.dll" target="@LIB_PREFIX@gthread-2.0@LIB_MIDDLE@0@LIB_SUFFIX@"/>
+  <dllmap dll="libgtk-win32-2.0-0.dll" target="@LIB_PREFIX@gtk-x11-2.0@LIB_MIDDLE@0@LIB_SUFFIX@"/>
+  <dllmap dll="gnome-desktop-2" target="@LIB_PREFIX@gnome-desktop-2@LIB_MIDDLE@11@LIB_SUFFIX@"/>
 </configuration>
--- origsrc/gnome-desktop-sharp-2.26.0/gnomepanel/glue/Makefile.am	2008-08-06 16:42:15.000000000 -0500
+++ src/gnome-desktop-sharp-2.26.0/gnomepanel/glue/Makefile.am	2009-12-09 02:51:56.745928800 -0600
@@ -10,7 +10,7 @@ libgnomepanelsharpglue_2_la_SOURCES = pa
 
 nodist_libgnomepanelsharpglue_2_la_SOURCES = generated.c
 
-libgnomepanelsharpglue_2_la_LDFLAGS = -module -avoid-version -no-undefined
+libgnomepanelsharpglue_2_la_LDFLAGS = -avoid-version -no-undefined
 
 libgnomepanelsharpglue_2_la_LIBADD = $(GNOME_PANEL_LIBS)
 
--- origsrc/gnome-desktop-sharp-2.26.0/gnomepanel/gnome-panel-sharp.dll.config.in	2008-08-27 14:04:35.000000000 -0500
+++ src/gnome-desktop-sharp-2.26.0/gnomepanel/gnome-panel-sharp.dll.config.in	2009-12-09 02:52:57.508034800 -0600
@@ -1,4 +1,4 @@
 <configuration>
-  <dllmap dll="libgobject-2.0-0.dll" target="libgobject-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
-  <dllmap dll="panel-applet-2" target="libpanel-applet-2@LIB_PREFIX@.0@LIB_SUFFIX@"/>
+  <dllmap dll="libgobject-2.0-0.dll" target="@LIB_PREFIX@gobject-2.0@LIB_MIDDLE@0@LIB_SUFFIX@"/>
+  <dllmap dll="panel-applet-2" target="@LIB_PREFIX@panel-applet-2@LIB_MIDDLE@0@LIB_SUFFIX@"/>
 </configuration>
--- origsrc/gnome-desktop-sharp-2.26.0/gnomeprint/gnome-print-sharp.dll.config.in	2008-08-04 09:07:08.000000000 -0500
+++ src/gnome-desktop-sharp-2.26.0/gnomeprint/gnome-print-sharp.dll.config.in	2009-12-09 02:53:26.992086200 -0600
@@ -1,5 +1,5 @@
 <configuration>
-  <dllmap dll="libgobject-2.0-0.dll" target="libgobject-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
-  <dllmap dll="gnomeprint-2-2" target="libgnomeprint-2-2@LIB_PREFIX@.0@LIB_SUFFIX@"/>
-  <dllmap dll="gnomeprintui-2-2" target="libgnomeprintui-2-2@LIB_PREFIX@.0@LIB_SUFFIX@"/>
+  <dllmap dll="libgobject-2.0-0.dll" target="@LIB_PREFIX@gobject-2.0@LIB_MIDDLE@0@LIB_SUFFIX@"/>
+  <dllmap dll="gnomeprint-2-2" target="@LIB_PREFIX@gnomeprint-2-2@LIB_MIDDLE@0@LIB_SUFFIX@"/>
+  <dllmap dll="gnomeprintui-2-2" target="@LIB_PREFIX@gnomeprintui-2-2@LIB_MIDDLE@0@LIB_SUFFIX@"/>
 </configuration>
--- origsrc/gnome-desktop-sharp-2.26.0/gtkhtml/glue/Makefile.am	2008-08-06 13:15:54.000000000 -0500
+++ src/gnome-desktop-sharp-2.26.0/gtkhtml/glue/Makefile.am	2009-12-09 02:51:59.772334100 -0600
@@ -10,7 +10,7 @@ libgtkhtmlsharpglue_2_la_SOURCES = # emp
 
 nodist_libgtkhtmlsharpglue_2_la_SOURCES = generated.c
 
-libgtkhtmlsharpglue_2_la_LDFLAGS = -module -avoid-version -no-undefined
+libgtkhtmlsharpglue_2_la_LDFLAGS = -avoid-version -no-undefined
 
 libgtkhtmlsharpglue_2_la_LIBADD = $(GTKHTML_LIBS)
 
--- origsrc/gnome-desktop-sharp-2.26.0/gtkhtml/gtkhtml-sharp.dll.config.in	2008-03-06 20:22:23.000000000 -0600
+++ src/gnome-desktop-sharp-2.26.0/gtkhtml/gtkhtml-sharp.dll.config.in	2009-12-09 02:51:21.302667000 -0600
@@ -1,4 +1,4 @@
 <configuration>
-  <dllmap dll="libgobject-2.0-0.dll" target="libgobject-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
-  <dllmap dll="gtkhtml-3.0" target="libgtkhtml-3.14@LIB_PREFIX@.19@LIB_SUFFIX@"/>
+  <dllmap dll="libgobject-2.0-0.dll" target="@LIB_PREFIX@gobject-2.0@LIB_MIDDLE@0@LIB_SUFFIX@"/>
+  <dllmap dll="gtkhtml-3.0" target="@LIB_PREFIX@gtkhtml-3.14@LIB_MIDDLE@19@LIB_SUFFIX@"/>
 </configuration>
--- origsrc/gnome-desktop-sharp-2.26.0/gtksourceview/glue/Makefile.am	2008-08-06 13:12:39.000000000 -0500
+++ src/gnome-desktop-sharp-2.26.0/gtksourceview/glue/Makefile.am	2009-12-09 02:52:03.500740600 -0600
@@ -10,7 +10,7 @@ libgtksourceview2sharpglue_2_la_SOURCES 
 
 nodist_libgtksourceview2sharpglue_2_la_SOURCES = generated.c
 
-libgtksourceview2sharpglue_2_la_LDFLAGS = -module -avoid-version -no-undefined
+libgtksourceview2sharpglue_2_la_LDFLAGS = -avoid-version -no-undefined
 
 libgtksourceview2sharpglue_2_la_LIBADD = $(GTKSOURCEVIEW_LIBS)
 
--- origsrc/gnome-desktop-sharp-2.26.0/gtksourceview/gtksourceview2-sharp.dll.config.in	2008-03-06 20:22:22.000000000 -0600
+++ src/gnome-desktop-sharp-2.26.0/gtksourceview/gtksourceview2-sharp.dll.config.in	2009-12-09 02:51:21.302667000 -0600
@@ -1,5 +1,5 @@
 <configuration>
-  <dllmap dll="libglib-2.0-0.dll" target="libglib-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
-  <dllmap dll="libgobject-2.0-0.dll" target="libgobject-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
-  <dllmap dll="gtksourceview-2.0" target="libgtksourceview-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
+  <dllmap dll="libglib-2.0-0.dll" target="@LIB_PREFIX@glib-2.0@LIB_MIDDLE@0@LIB_SUFFIX@"/>
+  <dllmap dll="libgobject-2.0-0.dll" target="@LIB_PREFIX@gobject-2.0@LIB_MIDDLE@0@LIB_SUFFIX@"/>
+  <dllmap dll="gtksourceview-2.0" target="@LIB_PREFIX@gtksourceview-2.0@LIB_MIDDLE@0@LIB_SUFFIX@"/>
 </configuration>
--- origsrc/gnome-desktop-sharp-2.26.0/nautilusburn/glue/Makefile.am	2008-08-06 13:01:32.000000000 -0500
+++ src/gnome-desktop-sharp-2.26.0/nautilusburn/glue/Makefile.am	2009-12-09 02:52:06.105945100 -0600
@@ -10,7 +10,7 @@ libnautilusburnsharpglue_2_la_SOURCES = 
 
 nodist_libnautilusburnsharpglue_2_la_SOURCES = generated.c
 
-libnautilusburnsharpglue_2_la_LDFLAGS = -module -avoid-version -no-undefined
+libnautilusburnsharpglue_2_la_LDFLAGS = -avoid-version -no-undefined
 
 libnautilusburnsharpglue_2_la_LIBADD = $(NAUTILUSBURN_LIBS)
 
--- origsrc/gnome-desktop-sharp-2.26.0/nautilusburn/nautilusburn-sharp.dll.config.in	2008-03-06 20:22:21.000000000 -0600
+++ src/gnome-desktop-sharp-2.26.0/nautilusburn/nautilusburn-sharp.dll.config.in	2009-12-09 02:51:21.318267100 -0600
@@ -1,3 +1,3 @@
 <configuration>
-  <dllmap dll="nautilus-burn" target="libnautilus-burn@LIB_PREFIX@.4@LIB_SUFFIX@"/>
+  <dllmap dll="nautilus-burn" target="@LIB_PREFIX@nautilus-burn@LIB_MIDDLE@4@LIB_SUFFIX@"/>
 </configuration>
--- origsrc/gnome-desktop-sharp-2.26.0/rsvg/rsvg2-sharp.dll.config.in	2008-03-06 20:22:21.000000000 -0600
+++ src/gnome-desktop-sharp-2.26.0/rsvg/rsvg2-sharp.dll.config.in	2009-12-09 02:51:21.318267100 -0600
@@ -1,3 +1,3 @@
 <configuration>
-  <dllmap dll="rsvg-2" target="librsvg-2@LIB_PREFIX@.2@LIB_SUFFIX@"/>
+  <dllmap dll="rsvg-2" target="@LIB_PREFIX@rsvg-2@LIB_MIDDLE@2@LIB_SUFFIX@"/>
 </configuration>
--- origsrc/gnome-desktop-sharp-2.26.0/vte/glue/Makefile.am	2008-03-06 20:22:21.000000000 -0600
+++ src/gnome-desktop-sharp-2.26.0/vte/glue/Makefile.am	2009-12-09 02:51:21.318267100 -0600
@@ -10,7 +10,7 @@ libvtesharpglue_2_la_SOURCES = # empty
 
 nodist_libvtesharpglue_2_la_SOURCES = generated.c
 
-libvtesharpglue_2_la_LDFLAGS = -module -avoid-version -no-undefined
+libvtesharpglue_2_la_LDFLAGS = -avoid-version -no-undefined
 
 libvtesharpglue_2_la_LIBADD = $(VTE_LIBS)
 
--- origsrc/gnome-desktop-sharp-2.26.0/vte/vte-sharp.dll.config.in	2008-03-06 20:22:21.000000000 -0600
+++ src/gnome-desktop-sharp-2.26.0/vte/vte-sharp.dll.config.in	2009-12-09 02:51:21.318267100 -0600
@@ -1,5 +1,6 @@
 <configuration>
-  <dllmap dll="libgobject-2.0-0.dll" target="libgobject-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
-  <dllmap dll="libatk-1.0-0.dll" target="libatk-1.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
-  <dllmap dll="vte" target="libvte@LIB_PREFIX@.9@LIB_SUFFIX@"/>
+  <dllmap dll="libgobject-2.0-0.dll" target="@LIB_PREFIX@gobject-2.0@LIB_MIDDLE@0@LIB_SUFFIX@"/>
+  <dllmap dll="libatk-1.0-0.dll" target="@LIB_PREFIX@atk-1.0@LIB_MIDDLE@0@LIB_SUFFIX@"/>
+  <dllmap dll="vte" target="@LIB_PREFIX@vte@LIB_MIDDLE@9@LIB_SUFFIX@"/>
+  <dllmap dll="vtesharpglue-2" target="@LIB_PREFIX@vtesharpglue-2@LIB_SUFFIX@"/>
 </configuration>
--- origsrc/gnome-desktop-sharp-2.26.0/wnck/glue/Makefile.am	2008-08-06 12:56:51.000000000 -0500
+++ src/gnome-desktop-sharp-2.26.0/wnck/glue/Makefile.am	2009-12-09 02:52:11.331954300 -0600
@@ -10,7 +10,7 @@ libwncksharpglue_2_la_SOURCES = # empty
 
 nodist_libwncksharpglue_2_la_SOURCES = generated.c
 
-libwncksharpglue_2_la_LDFLAGS = -module -avoid-version -no-undefined
+libwncksharpglue_2_la_LDFLAGS = -avoid-version -no-undefined
 
 libwncksharpglue_2_la_LIBADD = $(WNCK_LIBS)
 
--- origsrc/gnome-desktop-sharp-2.26.0/wnck/wnck-sharp.dll.config.in	2008-03-06 20:22:22.000000000 -0600
+++ src/gnome-desktop-sharp-2.26.0/wnck/wnck-sharp.dll.config.in	2009-12-09 02:51:21.333867100 -0600
@@ -1,4 +1,4 @@
 <configuration>
-  <dllmap dll="libgobject-2.0-0.dll" target="libgobject-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
-  <dllmap dll="wnck-1" target="libwnck-1@LIB_PREFIX@.22@LIB_SUFFIX@"/>
+  <dllmap dll="libgobject-2.0-0.dll" target="@LIB_PREFIX@gobject-2.0@LIB_MIDDLE@0@LIB_SUFFIX@"/>
+  <dllmap dll="wnck-1" target="@LIB_PREFIX@wnck-1@LIB_MIDDLE@22@LIB_SUFFIX@"/>
 </configuration>
